### DISCUSSION TITLE
Deleting Pull-Request

### DISCUSSION THREAD

**Discussion by TheMehranKhan** [community/community#28200] [Pull Requests] [answered]
Hey there Github Community
I wanted to ask is there a way for GitHub Support to delete a pull-request (made by me) on an inactive fork repo by an inactive user?
I just don't want that in my profile.

**Comment by jfullstackdev** [comment]
simply block the inactive user, and you will not be associated with that contributions

![image](https://user-images.githubusercontent.com/47092464/182647967-c9f848df-ff5d-4219-aad8-ad5dbfda0f67.png)

https://docs.github.com/en/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-personal-account

**Reply by TheMehranKhan** [reply]
Thanks for the reply @jdevstatic 
I blocked the user but I can still see this in my profile

<img width="489" alt="Screen Shot 2022-08-03 at 8 41 11 PM" src="https://user-images.githubusercontent.com/8843457/182657438-96f2deda-9ecb-4d09-aa7e-6a8adbc2dfc4.png">

**Reply by jfullstackdev** [reply]
what is the exact year of that ? I'm checking your timeline

**Reply by TheMehranKhan** [reply]
It's like 3-5 hours ago

**Comment by DenverCoder1** [comment, marked as answer]
Pull requests can't be deleted through the UI.

If you request, GitHub will delete pull requests when they contain sensitive information like passwords or other credentials. You can contact GitHub support here: https://support.github.com/contact.

**Reply by TheMehranKhan** [reply]
I thought about it too, already made a ticket but was trying to make sure if GitHub will delete a pull-request or not, Thanks @DenverCoder1

**Reply by jfullstackdev** [reply]
we are never sure of that but here in the Discussion, some Staff members might notice this and follow up, but again, there is no assurance

one thing also is that, it might be deleted on your timeline once it updates because you blocked the user, maybe 24 hours

also, if you are lucky, you can ask the owner of the fork to delete that fork, maybe you will be saved by that 

> Deleting a Repository 
> When you delete a GitHub repository, you delete it permanently with all its data, including the codebase, issues, pull requests, and everything else related to it.

https://rewind.com/blog/how-to-delete-github-repository/

**Reply by TheMehranKhan** [reply]
sure, thanks for the help anyways, hope I can get rid of it.

**Reply by queenofcorgis** [reply]
Hi there 👋🏼 @DenverCoder1 has it right - GitHub will delete pull requests, but only when they contain sensitive information.

**Reply by TheMehranKhan** [reply]
Thanks @queenofcorgis, Good news I contacted GitHub and it got resolved.
Appreciate your answers.

**Reply by queenofcorgis** [reply]
Wonderful - thanks for the update @TheMehranKhan ✅

**Comment by hinell** [comment]
ADD FEATURE TO DELETE UNMERGED PRS FOR THE GOD SAKE, I MADE A BUNCH OF PRS ACCIDENTALLY

THIS WILL HELP GREATLY KEEP YOUR DBS AWAY FROM TRASH

Thank you!

**Reply by ayushmorbar** [reply]
+1

**Comment by bennyschmidt** [comment]
Still can't delete PRs? Sometimes people in the open source world just open nonsense, it would be nice as a repo admin to be able to delete obvious spam.

**Reply by hinell** [reply]
@bennyschmidt Well it will be quickly abused...

**Reply by bennyschmidt** [reply]
@hinell Not having it allows for abuse, as I said in the previous comment. How do you think it would it be abused?

**Reply by hinell** [reply]
@bennyschmidt  Obvious spam should be deleted by the platform itself. My concern is that repo owners will delete useful PRs due to heavy elitism which would come at cost for the community. There are many cases of this syndrome in the open-source world sadly.

**Comment by mnaydenov** [comment]
Why not allow **DRAFT PRs** to be deletable!?!?

**Comment by mamift** [comment]
It is a bit weird that we can delete entire repos from our account, but not individual PRs. Heck, we can even delete our own [accounts](https://docs.github.com/en/account-and-profile/how-tos/account-management/deleting-your-personal-account), but somehow a red line is drawn around deleting individual pull requests? Makes no sense...

**Comment by mhaslinsky** [comment]
This is really a no brainer, just add it

**Comment by sid-harjeet** [comment]
Indeed, please allow users to abandon/delete their PRs , closing still keeps it as a record.  I don't understand why this can't be enabled. Why should my PR stay there forever when I decided to drop it for whatever reason .

**Comment by Marcono1234** [comment]
GitHub recently added the ability to "archive" a pull request ([changelog](https://github.blog/changelog/2026-07-16-repository-admins-can-archive-pull-requests/), [docs](https://docs.github.com/en/communities/moderating-comments-and-conversations/archive-pull-requests)).

This does not delete the pull request but restricts access to it to repository admins. However, archiving a pull request currently requires the repository admin role.

Note that this seems to be only about the pull request itself. The branch of the pull request might still be accessible, even after deletion or force-push through the [Activity view](https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository/using-the-activity-view-to-see-changes-to-a-repository). So if you want to redact sensitive data, you might still need to contact GitHub support.
